Assets/LeapMotionModules/UI/Scripts/BoxDepenetrator.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BoxDepenetrator : MonoBehaviour {
+
+  public SphereCollider sphere;
+
+  [Header("Automatic if null (this)")]
+  public BoxCollider box;
+
+  public struct DepenetrationRay {
+    public Vector3 position;
+    public Vector3 direction;
+  }
+  
+  private DepenetrationRay _depenetrationRay;
+  private bool _requiresDepenetrationThisFrame = false;
```

Comment:
Looks like the only reason this variable is not a local variable to update is that it is used in the Gizmos function.  I would recommend having a *separate* variable for use in the gizmos, something like 'didDoDepenetrationLastFrame' or something, and keep the 'requiresDepenetration' logic completely local to the Update function.